src/applications/vaos/new-appointment/components/TypeOfCarePage/index.jsx

```diff
@@ -92,7 +108,7 @@ function TypeOfCarePage({
           title="Type of care"
           schema={schema}
           uiSchema={uiSchema}
-          onSubmit={() => routeToNextAppointmentPage(history, pageKey)}
+          onSubmit={() => routeToNextAppointmentPage(history, pageKey, data)}
```

Comment:
One notable change is that we need to save the form data whenever we change pages, because any user updates are held in component state by default now.